### ISSUE TITLE
Does not support kubernetes secrets

### ISSUE THREAD

**Issue by mkantor** [vyshane/docker-compose-kubernetes#9]
Here's a simple way to see the issue:
1. Use [the `kube-up.sh` script](https://github.com/vyshane/docker-compose-kubernetes/blob/959c12de02c65e9850050f5f7c18ce63cf1b0710/kube-up.sh) to set up kubernetes.
2. Follow http://kubernetes.io/v1.1/docs/user-guide/secrets/README.html to create a secret and a pod that consumes it.
3. After creating the pod (and waiting some time for the wheels to spin), use `kubectl describe pods/secret-test-pod` to list the pod's events.

You will see events like these:

```
FailedSync   Error syncing pod, skipping: IsLikelyNotMountPoint("/var/lib/kubelet/pods/8c8d900f-992c-11e5-836b-46ca7ac5766f/volumes/kubernetes.io~secret/secret-volume"): file does not exist
FailedMount  Unable to mount volumes for pod "secret-test-pod_default": IsLikelyNotMountPoint("/var/lib/kubelet/pods/8c8d900f-992c-11e5-836b-46ca7ac5766f/volumes/kubernetes.io~secret/secret-volume"): file does not exist
```

And the pod will never make it out of the "Pending" state.

**Comment by mkantor** [comment]
I looked for upstream issues and found a few that seem relevant:
- https://github.com/kubernetes/kubernetes/issues/13424
- https://github.com/kubernetes/kubernetes/pull/13791
- https://github.com/kubernetes/kubernetes/issues/16055
- https://github.com/docker/docker/pull/17034

There are others too and it's hard to follow the tangled web of discussions (especially since I'm new to docker), but I _think_ this is not actually supported as of docker 1.9 due to limitations with how mounts are handled. I'd love to be proven wrong, though.

**Comment by mkantor** [comment]
```
$ docker version
Client:
 Version:      1.9.0
 API version:  1.21
 Go version:   go1.5.1
 Git commit:   76d6bc9
 Built:        Tue Nov  3 21:17:46 UTC 2015
 OS/Arch:      darwin/amd64

Server:
 Version:      1.9.0
 API version:  1.21
 Go version:   go1.4.3
 Git commit:   76d6bc9
 Built:        Tue Nov  3 19:20:09 UTC 2015
 OS/Arch:      linux/amd64
```

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"0", GitVersion:"v1.0.6", GitCommit:"388061f00f0d9e4d641f9ed4971c775e1654579d", GitTreeState:"clean"}
Server Version: version.Info{Major:"1", Minor:"1", GitVersion:"v1.1.2", GitCommit:"3085895b8a70a3d985e9320a098e74f545546171", GitTreeState:"clean"}
```

**Comment by vyshane** [comment]
@mkantor You are right, I'm also currently waiting for this to be fixed upstream.

Unfortunately for now I've resorted to using alternative solutions for local development. For example: [Kubernetes on CoreOS via Vagrant](https://github.com/coreos/coreos-kubernetes/tree/master/single-node).

**Comment by huggsboson** [comment]
Looks like this merged:
https://github.com/docker/docker/pull/17034

Does that unblock this?

If you're doing kube on coreos on vagrant, do you like that path overall better than docker-machine + docker-compose-kube?

**Comment by vyshane** [comment]
> Looks like this merged:
> docker/docker#17034
> 
> Does that unblock this?

@huggsboson my guess is that Kubernetes will need to make use of this when Docker cuts a release that contains the new feature.

In the mean time, Kubernetes was supposed to have a workaround in kubernetes/kubernetes#17273. However, I'm currently seeing the same problem that is described in kubernetes/kubernetes#18513 when I use the newly-minted hyperkube v1.1.3.

So, unfortunately, I'm still waiting for a solution that works.

> If you're doing kube on coreos on vagrant, do you like that path overall better than docker-machine + docker-compose-kube?

I'm partial to a dockerised setup for development because I get very fast Kubernetes setups and tear downs. This is very useful when iterating quickly.

**Comment by huggsboson** [comment]
I spent a while playing with kubernetes/kubernetes#19069.  It does, in fact, fix the secrets mounting issue.  Playing with that branch had me using the new hyperkube image.  That image itself spins up a kube-proxy and etcd.  So the only dependencies you'll need to start for yourself at that point are skydns.

Between that and using something similar to:

```
kubectl config set-cluster docker-machine-default --server=$(docker-machine ip default):443 --certificate-authority=${PWD}/ssl/ca.pem
kubectl config set-credentials vagrant-admin --certificate-authority=${PWD}/ssl/ca.pem --client-key=${PWD}/ssl/admin-key.pem --client-certificate=${PWD}/ssl/admin.pem
kubectl config set-context vagrant --cluster=vagrant --user=vagrant-admin
kubectl config use-context vagrant
```

from: https://coreos.com/kubernetes/docs/latest/kubernetes-on-vagrant-single.html

We might be able to really slim down the number of moving parts.

**Comment by huggsboson** [comment]
Not sure how long it will take for upstream to land.  I may try copying their fix down and submitting a pull request, it's basically:
nsenter --target=1 --mount --wd=. -- ./hyperkube kubelet $@

Then remove --containerized.

**Comment by vyshane** [comment]
@huggsboson thanks for doing this. I've merged your PR.
